### PR TITLE
Fix mark.it titles when used with mark.parametrize

### LINKED ISSUE #22: Incompatibility between pytest.mark.it and pytest.mark.parametrize
I've been playing around with the new `pytest.mark.it` and `pytest.mark.describe` features, and they're wonderful (thanks for implementing those), but I have hit an issue with regard to parametrization.

When parametrizing a test, with `pytest.mark.parametrize`, the test is repeated several times with different parameters, and formats the different parameter ids in square brackets, e.g.:

```cmd
 [x] from authentication provider boundary case transport name[ALL CAPS]
 [x] from authentication provider boundary case transport name[mIxEd CaSe]
```

However, when using the name overrides, it overrides, the square bracket parameter ids are lost, and the name given by `pytest.mark.it` is just simply repeated twice:

```cmd
 [x] from_authentication_provider boundary case transport_name
 [x] from_authentication_provider boundary case transport_name
```
Ideally, I can use name overrides while keeping parameter ids.

### PR DESCRIPTION
**Pull request by RomuloOliveira**
Allow use of custom titles using `pytest.mark.it` along with parametrized tests. Fixes #22.

When `mark.parametrize` is used, pytest includes the parameters in the last part of `nodeid` (e.g. `test_module_name.py::test_a_passing_test[this is a param]`).
I used the fact that function names cannot contain brackets to safely extract parameters. Parameters may include brackets, though.

I think function names and arguments I wrote may need a more thorough review.

### CODE REVIEW

**Comment by RomuloOliveira** [comment]
@cartertinney could you please test if this will do the job for the cases you're working on? You can test using `pip install git+https://github.com/renanivo/pytest-testdox.git@ro-mark-it-parametrize-fix`.

@renanivo I've marked as a draft pull request until you or @cartertinney can confirm that it work as expected.

Here the output for some tests I did:

```python
    @pytest.mark.parametrize('a_param', [
        'String',
        1234,
        [],
        {}
    ])
    @pytest.mark.it('It should display a nice name')
    def test_foobar(self, a_param):
        assert True
```

```
Include Parametrized
 ✓ It should display a nice name[String]
 ✓ It should display a nice name[1234]
 ✓ It should display a nice name[a_param2]
 ✓ It should display a nice name[a_param3]
```

Without using `pytest-testdox`, this is the output with `-vv`:

```
tests/test_formatters.py::TestIncludeParametrized::test_foobar[String] PASSED                                              [ 25%]
tests/test_formatters.py::TestIncludeParametrized::test_foobar[1234] PASSED                                                [ 50%]
tests/test_formatters.py::TestIncludeParametrized::test_foobar[a_param2] PASSED                                            [ 75%]
tests/test_formatters.py::TestIncludeParametrized::test_foobar[a_param3] PASSED                                            [100%]
```

File: tests/test_plugin.py [hunk 1]
```diff
@@ -174,3 +174,41 @@ def test_foo(self):
...
+            @pytest.mark.{}('should pass with parameters')
+            def test_a_passing_test(par):
+                assert True
+        """.format(
+            constants.TITLE_MARK
+        ))
+
+        result = testdir.runpytest('--testdox')
+
+        assert 'should pass with parameters[param' in result.stdout.str()
```
Comment (renanivo) [inline tests/test_plugin.py]: What about using one `assert` for each parameter?
```suggestion
        assert 'should pass with parameters[param1]' in result.stdout.str()
        assert 'should pass with parameters[param2]' in result.stdout.str()
```

File: tests/test_plugin.py [hunk 2]
```diff
@@ -174,3 +174,41 @@ def test_foo(self):
...
+            @pytest.mark.parametrize('par', ['param1', 'param2'])
+            def test_a_passing_test(par):
+                assert True
+        """.format(
+            constants.TITLE_MARK
+        ))
+
+        result = testdir.runpytest('--testdox')
+
+        assert 'should pass with parameters[param' in result.stdout.str()
```
Comment (renanivo) [inline tests/test_plugin.py]: Same here
```suggestion
        assert 'should pass with parameters[param1]' in result.stdout.str()
        assert 'should pass with parameters[param2]' in result.stdout.str()
```

**Comment by cartertinney** [comment]
Yeah, just tried it out, working great here. Thanks a bunch.
